syntax = "proto3";

option cc_generic_services = true;

package platform;

// 心跳消息
message msg_heartbeat
{
	int64		time = 1;  // time
}

// 连接成功消息
message msg_connected
{
	int32		client_id = 1;	// client_id
	int32		state = 2;  // state
	int64		time = 3;  // time
}

// 登录消息
message msg_login_request
{
	int32	device		= 1;	// device
	string  username	= 2;	// username
	string  userpsd		= 3;	// userpsd
}

message msg_login_reponse
{
	string  msg			= 1;	// msg
}

message msg_game_start_reponse
{
	int32    state = 1;		// state
}

message msg_game_end_reponse
{
	int32    state = 1;		// state
}

message msg_situp_request
{
	int32    uid = 1;		// uid
	int32    seat = 2;		// seat
	bool     lookup = 3;	// lookup
}

message msg_situp_reponse
{
	int32    uid = 1;		// uid
	int32    seat = 2;		// seat
	bool     lookup = 3;	// lookup
}

message msg_sitdown_request
{
	int32    uid = 1;		// uid
	int32    seat = 2;		// seat
	bool     lookup = 3;	// lookup
}

message msg_sitdown_reponse
{
	int32    uid = 1;		// uid
	int32    seat = 2;		// seat
	bool     lookup = 3;	// lookup
}

message EchoReqMsg
{
	int32     a = 1;			// a
	int32     b = 2;			// b
}

message EchoRespMsg
{
	int32     c = 1;			// c
	string    msg = 2;			// msg
}

service MyService {
	rpc Echo(EchoReqMsg) returns(EchoRespMsg);
}